File: src/Utilities/KeyTraits.cc
//---------------------------------Spheral++----------------------------------//
// KeyTraits
//
// Encapsulate how we think about keys for the space filling curves.
// This is specialized assuming we're working with 64 bit unsigned long long.
//
// Created by JMO, Wed Apr  9 13:21:29 PDT 2008
//----------------------------------------------------------------------------//
#include "KeyTraits.hh"

namespace Spheral {

const uint32_t KeyTraits::numbits = 64;
const uint32_t KeyTraits::numbits1d = 20;
const KeyTraits::Key KeyTraits::zero = 0ULL;
const KeyTraits::Key KeyTraits::one = 1ULL;
const KeyTraits::Key KeyTraits::two = 2ULL;
const KeyTraits::Key KeyTraits::maxKey1d = 2097152ULL;
const KeyTraits::Key KeyTraits::maxKey = 9223372036854775808ULL;

}

File: src/Pybind11Wraps/Utilities/KeyTraits.py
#-------------------------------------------------------------------------------
# KeyTraits
#-------------------------------------------------------------------------------
from PYB11Generator import *

class KeyTraits:
    """Encapsulate how we think about keys for the space filling curves.
This is specialized assuming we're working with 64 bit uint64_t."""

    numbits =   PYB11readonly(static=True, doc="Total number of bits in a Key", returnpolicy="copy")
    numbits1d = PYB11readonly(static=True, doc="Number of bits allowed per dimension in a Key", returnpolicy="copy")
    zero =      PYB11readonly(static=True, doc="Null value", returnpolicy="copy")
    one =       PYB11readonly(static=True, doc="Unit value", returnpolicy="copy")
    two =       PYB11readonly(static=True, doc="Like, the two value", returnpolicy="copy")
    maxKey1d =  PYB11readonly(static=True, doc="The maximum value in a dimension", returnpolicy="copy")
    maxKey =    PYB11readonly(static=True, doc="The maximum possible key", returnpolicy="copy")
